src/ext/java/org/opentripplanner/ext/emissions/Co2EmissionsDataReader.java

```diff
@@ -92,35 +92,53 @@ private Map<FeedScopedId, Double> readEmissions(InputStream stream, String feedI
       String avgCo2PerVehiclePerKmString = reader.get("avg_co2_per_vehicle_per_km");
       String avgPassengerCountString = reader.get("avg_passenger_count");
 
-      if (avgCo2PerVehiclePerKmString.isEmpty()) {
+      if (!exists(routeId)) {
         issueStore.add(
           "InvalidEmissionData",
-          "Value for avg_co2_per_vehicle_per_km is missing in the Emissions.txt for route %s",
+          "Value for routeId is missing in the emissions.txt for line: %s.",
+          reader.getRawRecord()
+        );
+      }
+      if (!exists(avgCo2PerVehiclePerKmString)) {}
+      {
+        issueStore.add(
+          "InvalidEmissionData",
+          "Value for avg_co2_per_vehicle_per_km is missing in the emissions.txt for route %s",
           routeId
         );
       }
-      if (avgPassengerCountString.isEmpty()) {
+      if (!exists(avgPassengerCountString)) {
         issueStore.add(
           "InvalidEmissionData",
-          "Value for avg_passenger_count is missing in the Emissions.txt for route %s",
+          "Value for avg_passenger_count is missing in the emissions.txt for route %s",
           routeId
         );
       }
-
-      Double avgCo2PerVehiclePerMeter = Double.parseDouble(avgCo2PerVehiclePerKmString) / 1000;
-      Double avgPassengerCount = Double.parseDouble(reader.get("avg_passenger_count"));
-      Optional<Double> emissions = calculateEmissionsPerPassengerPerMeter(
-        routeId,
-        avgCo2PerVehiclePerMeter,
-        avgPassengerCount
-      );
-      if (emissions.isPresent()) {
-        emissionsData.put(new FeedScopedId(feedId, routeId), emissions.get());
+      if (
+        exists(feedId) &&
+        exists(routeId) &&
+        exists(avgCo2PerVehiclePerKmString) &&
+        exists(avgPassengerCountString)
+      ) {
+        Double avgCo2PerVehiclePerMeter = Double.parseDouble(avgCo2PerVehiclePerKmString) / 1000;
+        Double avgPassengerCount = Double.parseDouble(reader.get("avg_passenger_count"));
+        Optional<Double> emissions = calculateEmissionsPerPassengerPerMeter(
+          routeId,
+          avgCo2PerVehiclePerMeter,
+          avgPassengerCount
+        );
+        if (emissions.isPresent()) {
+          emissionsData.put(new FeedScopedId(feedId, routeId), emissions.get());
+        }
       }
     }
     return emissionsData;
   }
 
+  private boolean exists(String s) {
```

Comment:
You can replace this with `StringUtils.hasValue`.

Reply:
Of course... fixed